### ISSUE TITLE
BUG: `genfromtxt` silently returns nan whenever reading complex values

### ISSUE THREAD

**Issue by Rodrigo-Tenorio** [numpy/numpy#24900]
### Describe the issue:

Whenever reading complex numbers using `genfromtxt`, numpy silently returns `nan`.

I've seen similar issues discussed in other places (e.g. https://github.com/numpy/numpy/issues/7895),
but I haven't found any issue for this specific case.

### Reproduce the code example:

```python
import numpy as np

data = np.array([0 + 0j, 1 + 1j])
np.savetxt(outfile := "./array_in_file.txt", data)
loaded_data = np.genfromtxt(outfile)

print(data)
print(loaded_data)
```

### Error message:

```shell
[0.+0.j 1.+1.j]
[nan nan]
```

### Runtime information:

>>> import sys, numpy; print(numpy.__version__); print(sys.version)
1.26.0
3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0]

>>> print(numpy.show_runtime())
WARNING: `threadpoolctl` not found in system! Install it by `pip install threadpoolctl`. Once installed, try `np.show_runtime` again for more detailed build information
[{'numpy_version': '1.26.0',
  'python': '3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0]',
  'uname': uname_result(system='Linux', node='pop-os', release='6.2.6-76060206-generic', version='#202303130630~1683753207~22.04~77c1465 SMP PREEMPT_DYNAMIC Wed M', machine='x86_64')},
 {'simd_extensions': {'baseline': ['SSE', 'SSE2', 'SSE3'],
                      'found': ['SSSE3',
                                'SSE41',
                                'POPCNT',
                                'SSE42',
                                'AVX',
                                'F16C',
                                'FMA3',
                                'AVX2'],
                      'not_found': ['AVX512F',
                                    'AVX512CD',
                                    'AVX512_KNL',
                                    'AVX512_KNM',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL']}}]
None

### Context for the issue:

The current behavior is to silently parse a "numerical value" as a `nan`.
I understand this is the default case for other non-numerical values such as strings,
but I would have expected complex numbers to behave more closely to numbers.

I'm unaware of the complexity of this function, but would it be useful to add a minor
warning for the case of parsing complex numbers as `nan`?

**Comment by rossbar** [comment]
> For me it's not a bug as numpy should not guess if it's a string or a complex number.

I agree, though in this case the dtype-determining logic lands on float which is arguably also not correct. Perhaps a better solution would be to raise, but I'm not sure how difficult that is without breaking the rest of the dtype-guessing logic.

**Comment by timrdmr** [comment]
Hello, I think I found the issue. The `dtype` parameter is set to `float` by default: https://github.com/numpy/numpy/blob/18c6157f5da5736575b6d8d492aacca3b5d69551/numpy/lib/_npyio_impl.py#L1713

When I call the above code with `dtype=None`, `genfromtxt` correctly determines the complex type. The type detection also works with float. So, I see no reason why `float` is the default here. Additionally, this is also documented: "If None, the dtypes will be determined by the contents of column, individually."

I propose to change the default of the parameter to `None`. What do you think?
It would be very nice if I could provide a fix.

**Comment by rossbar** [comment]
Ok - my comment above was based on an incorrect assumption that the default dtype for `genfromtext` was `dtype=None`. As pointed out in #25301, that is not the case. Given this, I don't think the issue raised in the OP is a bug --- the default `dtype` is `float` so genfromtext is doing exactly what is requested: trying to interpreted the input tokens as floats. It is the `loose=True` (also the default) that results in the NaN's. This is also the correct behavior.

I'm inclined to close this issue. We can open a separate one about changing the default dtype parameter to `None`, but AFAICT this is not a bug.

**Comment by mattip** [comment]
In a recent triage meeting we would like to propose that the default for `loose` should be `False` by default via a deprecation cycle. It would be good if the error message mentioned "you can avoid the error if you want by setting `loose=True`".
